src/_igraph/operators.c

```diff
@@ -0,0 +1,317 @@
+    Py_DECREF(it);
+    igraph_vector_ptr_destroy(&gs);
+    return NULL;
+  }
+  Py_DECREF(it);
+  no_of_graphs = (long int) igraph_vector_ptr_size(&gs);
+
+  /* Create disjoint union */
+  if (igraph_disjoint_union_many(&g, &gs)) {
+    igraph_vector_ptr_destroy(&gs);
+    igraphmodule_handle_igraph_error();
+    return NULL;
+  }
+
+  igraph_vector_ptr_destroy(&gs);
+
+  /* this is correct as long as attributes are not copied by the
+   * operator. if they are copied, the initialization should not empty
+   * the attribute hashes */
+  if (no_of_graphs > 0) {
+    result = igraphmodule_Graph_subclass_from_igraph_t(
+        result_type,
+        &g);
+  }
+  else {
+    result = igraphmodule_Graph_from_igraph_t(&g);
+  }
+
+  return result;
+}
+
+
+/** \ingroup python_interface_graph
+ * \brief Creates the union of two or more graphs
+ */
+PyObject *igraphmodule__union(PyObject *self,
+		PyObject *args, PyObject *kwds)
+{
+  static char* kwlist[] = { "graphs", "edgemaps", NULL };
+  PyObject *it, *em_list, *graphs, *with_edgemaps_o;
+  int with_edgemaps = 0;
+  long int no_of_graphs;
+  igraphmodule_GraphObject *o;
+  PyObject *result;
+  PyTypeObject *result_type;
+  igraph_t g;
+
+  if (!PyArg_ParseTupleAndKeywords(args, kwds, "O|O", kwlist,
+      &graphs, &with_edgemaps_o))
+    return NULL;
+
+  if (PyObject_IsTrue(with_edgemaps_o))
+    with_edgemaps = 1;
+
+  /* Needs to be an iterable */
+  it = PyObject_GetIter(graphs);
+  if (!it) {
+      Py_DECREF(it);
+      return igraphmodule_handle_igraph_error();
+  }
+
+  /* Get all elements, store the graphs in an igraph_vector_ptr */
+  igraph_vector_ptr_t gs;
+  if (igraph_vector_ptr_init(&gs, 0)) {
+    Py_DECREF(it);
+    return igraphmodule_handle_igraph_error();
+  }
+  if (igraphmodule_append_PyIter_of_graphs_to_vector_ptr_t_with_type(it, &gs, &result_type)) {
+    Py_DECREF(it);
+    igraph_vector_ptr_destroy(&gs);
+    return NULL;
+  }
+  Py_DECREF(it);
+  no_of_graphs = (long int) igraph_vector_ptr_size(&gs);
+
+  if (with_edgemaps) {
+    /* prepare edgemaps */
+    igraph_vector_ptr_t edgemaps;
+    if (igraph_vector_ptr_init(&edgemaps, 0)) {
+      return igraphmodule_handle_igraph_error();
+    }
+
+    /* Create union */
+    if (igraph_union_many(&g, &gs, &edgemaps)) {
+      igraph_vector_ptr_destroy(&gs);
+      igraph_vector_ptr_destroy(&edgemaps);
+      igraphmodule_handle_igraph_error();
+      return NULL;
+    }
+
+    /* extract edgemaps */
+    long int i;
+    em_list = PyList_New((Py_ssize_t) no_of_graphs);
+    for (i = 0; i < no_of_graphs; i++) {
+      long int j;
+      long int no_of_edges = (long int) igraph_ecount(VECTOR(gs)[i]);
+      igraph_vector_t *map = VECTOR(edgemaps)[i];
+      PyObject *emi = PyList_New((Py_ssize_t) no_of_edges);
+      for (j = 0; j < no_of_edges; j++) {
+        PyObject *dest = PyLong_FromLong(VECTOR(*map)[j]);
+        PyList_SetItem(emi, (Py_ssize_t) j, dest);
```

Comment:
You can use `PyList_SET_ITEM()` when initializing newly created lists; it is slightly faster than `PyList_SetItem`. `PyList_SET_ITEM()` will automatically steal a reference to `dest`.

If you don't use `PyList_SET_ITEM()` and stick to `PyList_SetItem`, then you will need to drop your own reference to `dest` because `PyList_SetItem` calls `Py_INCREF()` on `dest`.

Reply:
yep, thanks for the tip!